crates/sui-rest-api/src/transactions/resolve.rs

```diff
@@ -0,0 +1,599 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+
+use super::execution::SimulateTransactionQueryParameters;
+use super::TransactionSimulationResponse;
+use crate::accept::AcceptFormat;
+use crate::objects::ObjectNotFoundError;
+use crate::openapi::ApiEndpoint;
+use crate::openapi::OperationBuilder;
+use crate::openapi::RequestBodyBuilder;
+use crate::openapi::ResponseBuilder;
+use crate::openapi::RouteHandler;
+use crate::reader::StateReader;
+use crate::response::ResponseContent;
+use crate::RestError;
+use crate::RestService;
+use crate::Result;
+use axum::extract::Query;
+use axum::extract::State;
+use axum::Json;
+use itertools::Itertools;
+use move_binary_format::normalized;
+use schemars::JsonSchema;
+use sui_protocol_config::ProtocolConfig;
+use sui_sdk_types::types::Argument;
+use sui_sdk_types::types::Command;
+use sui_sdk_types::types::ObjectId;
+use sui_sdk_types::types::Transaction;
+use sui_sdk_types::types::UnresolvedInputArgument;
+use sui_sdk_types::types::UnresolvedObjectReference;
+use sui_sdk_types::types::UnresolvedProgrammableTransaction;
+use sui_sdk_types::types::UnresolvedTransaction;
+use sui_types::base_types::ObjectID;
+use sui_types::base_types::ObjectRef;
+use sui_types::base_types::SuiAddress;
+use sui_types::effects::TransactionEffectsAPI;
+use sui_types::gas::GasCostSummary;
+use sui_types::gas_coin::GasCoin;
+use sui_types::move_package::MovePackage;
+use sui_types::transaction::CallArg;
+use sui_types::transaction::GasData;
+use sui_types::transaction::ObjectArg;
+use sui_types::transaction::ProgrammableTransaction;
+use sui_types::transaction::TransactionData;
+use sui_types::transaction::TransactionDataAPI;
+use tap::Pipe;
+
+// TODO
+// - Updating the UnresolvedTransaction format to provide less information about inputs
+// - handle basic type inference and BCS serialization of pure args
+pub struct ResolveTransaction;
+
+impl ApiEndpoint<RestService> for ResolveTransaction {
+    fn method(&self) -> axum::http::Method {
+        axum::http::Method::POST
+    }
+
+    fn path(&self) -> &'static str {
+        "/transactions/resolve"
+    }
+
+    fn operation(
+        &self,
+        generator: &mut schemars::gen::SchemaGenerator,
+    ) -> openapiv3::v3_1::Operation {
+        OperationBuilder::new()
+            .tag("Transactions")
+            .operation_id("ResolveTransaction")
+            .query_parameters::<ResolveTransactionQueryParameters>(generator)
+            .request_body(
+                RequestBodyBuilder::new()
+                    // .json_content::<UnresolvedTransaction>(generator)
```

Comment:
I guess the TODO to add the JSON support is why this is commented out. Maybe add a TODO here too.

Reply:
its already covered by the TODO above